updater/lib/dependabot/dependency_group_engine.rb

```diff
@@ -41,6 +48,28 @@ def self.from_job_config(job:)
       new(dependency_groups: groups)
     end
 
+    sig { params(job: Dependabot::Job).void }
+    def self.validate_group_configuration!(job)
+      return unless job.dependency_groups.any?
+
+      unsupported_groups = job.dependency_groups.select do |group|
+        rules = group["rules"] || {}
+        rules.key?("dependency-type") &&
+          !PACKAGE_MANAGERS_SUPPORTING_DEPENDENCY_TYPE.include?(job.package_manager)
+      end
+
+      return unless unsupported_groups.any?
+
+      group_names = unsupported_groups.map { |g| g["name"] }.join(", ")
+      Dependabot.logger.warn <<~WARN
+        The 'dependency-type' option is not supported for the '#{job.package_manager}' package manager.
+        It is only supported for: #{PACKAGE_MANAGERS_SUPPORTING_DEPENDENCY_TYPE.join(', ')}.
+        Affected groups: #{group_names}
+
+        This option will be ignored. Please remove it from your configuration or use a supported package manager.
+      WARN
+    end
```

Comment:
I am late as this is already merged but I think that just a warn is not what my issue requested. I was hoping that the job would fail fast instead 

Logging this as a warning makes it just as hard for the user to find out as we should not expect our users to read the job logs in most situations